x/fdsmr/node.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fdsmr
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/set"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/x/dsmr"
+)
+
+type Bonder[T dsmr.Tx] interface {
+	// Bond returns if a transaction can be built into a chunk by this node.
+	// If this returns true, Unbond is guaranteed to be called.
+	Bond(tx T) bool
+	// Unbond is called when a tx from an account either expires or is accepted.
```

Comment:
Do we guarantee that `Bond` was called on this tx previously? If so, could we add that to the comment

Reply:
This is referenced the comment on `Bond`, but I can add a similar note to `Unbond`